src/re_frame/subs.cljc

```diff
@@ -301,23 +304,30 @@
          (let [subscriptions (inputs-fn query-vec)
                reaction-id   (atom nil)
                reaction      (make-reaction
-                               (fn [] (trace/with-trace {:operation (first-in-vector query-vec)
-                                                         :op-type   :sub/run
-                                                         :tags      {:query-v  query-vec
-                                                                     :reaction @reaction-id}}
-                                        (computation-fn (deref-input-signals subscriptions query-id) query-vec))))]
+                               (fn []
+                                 (let [sub-value (computation-fn (deref-input-signals subscriptions query-id) query-vec)]
+                                   (trace/with-trace {:operation (first-in-vector query-vec)
+                                                      :op-type   :sub/run
+                                                      :tags      {:sub-value  sub-value
+                                                                  :query-v    query-vec
+                                                                  :reaction   @reaction-id}}
+                                                     sub-value))))]
+
            (reset! reaction-id (reagent-id reaction))
            reaction))
         ([db query-vec dyn-vec]
          (let [subscriptions (inputs-fn query-vec dyn-vec)
                reaction-id   (atom nil)
                reaction      (make-reaction
                                (fn []
-                                 (trace/with-trace {:operation (first-in-vector query-vec)
-                                                    :op-type   :sub/run
-                                                    :tags      {:query-v  query-vec
-                                                                :dyn-v    dyn-vec
-                                                                :reaction @reaction-id}}
-                                   (computation-fn (deref-input-signals subscriptions query-id) query-vec dyn-vec))))]
+                                 (let [sub-value (computation-fn (deref-input-signals subscriptions query-id) query-vec dyn-vec)]
```

Comment:
The sub-value needs to be calculated within the `with-trace` so that we can capture accurate timings for it. I think you might need to use `trace/with-trace` to start the trace, calculate the sub value in a let, then use merge-trace to add it to the trace.

Reply:
that makes sense, I will move it so that it is calculated within the `with-trace`